src/features/session_replay/aggregate/index.js

```diff
@@ -223,8 +223,27 @@ export class Aggregate extends AggregateBase {
   getHarvestContents () {
     const agentRuntime = getRuntime(this.agentIdentifier)
     const info = getInfo(this.agentIdentifier)
-    const firstTimestamp = this.timestamp.event.first || this.timestamp.cycle.first
-    const lastTimestamp = this.timestamp.event.last || this.timestamp.cycle.last
+
+    // do not let the last node be a meta node, since this NEEDS to precede a snapshot
+    // we will manually inject it later if we find a payload that is missing a meta node
+    const payloadEndsWithMeta = this.events[this.events.length - 1]?.type === 4
+    if (payloadEndsWithMeta) {
+      this.hasMeta = false
```

Comment:
Could the payload still have meta nodes even if the last node is removed?

Reply:
good point, i think it is possible yes.  I will revise this